WordPress/Classes/Extensions/MediaLibrary+Local.swift

```diff
@@ -6,34 +6,53 @@ extension MediaLibrary {
 
     fileprivate static let mediaDirectoryName = "Media"
 
+    /// Type of the local Media directory URL in implementation.
+    ///
+    enum MediaDirectoryType {
+        case uploads // Default, system Documents directory, for persisting media files for upload.
+        case cache // System Caches directory, for creating discardable media files, such as thumbnails.
+        case temporary // System temporary directory, used for unit testing or temporary media files.
+    }
+
     // MARK: - Class methods
 
     /// Returns filesystem URL for the local Media directory.
     ///
-    class func localDirectory() throws -> URL {
+    class func localDirectory(_ type: MediaDirectoryType) throws -> URL {
         let fileManager = FileManager.default
-        let documents = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
-        var media = documents.appendingPathComponent(mediaDirectoryName, isDirectory: true)
-
+        // Get a parent directory, based on the type.
+        let parentDirectory: URL
+        if type == .cache {
+            parentDirectory = fileManager.urls(for: .cachesDirectory, in: .userDomainMask).first!
+        } else if type == .temporary {
+            parentDirectory = fileManager.temporaryDirectory
+        } else {
+            parentDirectory = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+        }
+        let mediaDirectory = parentDirectory.appendingPathComponent(mediaDirectoryName, isDirectory: true)
         // Check whether or not the file path exists for the Media directory.
         // If the filepath does not exist, or if the filepath does exist but it is not a directory, try creating the directory.
         // Note: This way, if unexpectedly a file exists but it is not a dir, an error will throw when trying to create the dir.
         var isDirectory: ObjCBool = false
-        if fileManager.fileExists(atPath: media.path, isDirectory: &isDirectory) == false || isDirectory.boolValue == false {
-            try fileManager.createDirectory(at: media, withIntermediateDirectories: true, attributes: nil)
-            var resourceValues = URLResourceValues()
-            resourceValues.isExcludedFromBackup = true
-            try media.setResourceValues(resourceValues)
+        if fileManager.fileExists(atPath: mediaDirectory.path, isDirectory: &isDirectory) == false || isDirectory.boolValue == false {
+            try fileManager.createDirectory(at: mediaDirectory, withIntermediateDirectories: true, attributes: nil)
```

Comment:
With this implementation, both the caches and tmp directories won't back up, or iOS will disregard the contents as it sees fit.

For the uploads directory, I avoided explicitly disabling backups as we technically want to hang on to these contents for any pending uploads.

If the user were to have a couple of local posts with Media ready for upload, but restored their phone from a backup for whatever reason, we'll still have the pending files. That was the idea anyways, do you think we should go ahead and exclude from backup regardless? @SergioEstevao